### PR TITLE
Refactor control flow validation, allow break in switches

### PR DESCRIPTION
**Pull request by kvark**
Helps #940

### CODE REVIEW

File: src/valid/function.rs [hunk 1]
```diff
@@ -325,24 +333,37 @@ impl super::Validator {
...
+                    if !context.abilities.contains(ControlFlowAbility::CONTINUE) {
+                        return Err(FunctionError::ContinueOutsideOfLoop);
                     }
                     finished = true;
                 }
                 S::Return { value } => {
-                    if !context.flags.contains(Flags::CAN_JUMP) {
+                    if !context.abilities.contains(ControlFlowAbility::RETURN) {
                         return Err(FunctionError::InvalidReturnSpot);
                     }
```
Comment (jimblandy) [inline src/valid/function.rs:368]: Are `Kill` statements permitted in `Loop` `continuing` blocks?

**Review by jimblandy — APPROVED** [review]
Looks good to me.

File: src/valid/function.rs (hunk 1 above)
Comment (kvark) [reply src/valid/function.rs:368]: I assume it is

**Comment by kvark** [comment]
Thank you for the review!

**Pull request merged**